gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/avro/AvroKeyCombineFileInputFormat.java

```diff
@@ -0,0 +1,141 @@
+/* (c) 2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.compaction.mapreduce.avro;
+
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.util.HadoopUtils;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.mapred.AvroKey;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.NullWritable;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.Job;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.hadoop.mapreduce.RecordReader;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.apache.hadoop.mapreduce.lib.input.CombineFileInputFormat;
+import org.apache.hadoop.mapreduce.lib.input.CombineFileRecordReader;
+import org.apache.hadoop.mapreduce.lib.input.CombineFileSplit;
+
+import com.google.common.collect.Lists;
+
+
+/**
+ * A subclass of {@link org.apache.hadoop.mapreduce.lib.input.CombineFileInputFormat}. This class is able to
+ * handle the case where the input path has subdirs which contain data files, which is not the case with
+ * {@link org.apache.hadoop.mapreduce.lib.input.CombineFileInputFormat}.
+ *
+ * @author ziliu
+ */
+public class AvroKeyCombineFileInputFormat extends CombineFileInputFormat<AvroKey<GenericRecord>, NullWritable> {
```

Comment:
Two things:
* the javadoc implies that this class is for all recursive combine file input formats, but the class specifically only looks at avro files. Javadoc needs to be corrected.
* the class name should specify that this input format is recursive.